llvm/test/CodeGen/AMDGPU/s_uaddo_pseudo.ll

```diff
@@ -0,0 +1,28 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 6
+; RUN: llc -mtriple=amdgcn-amd-amdpal -stop-after=amdgpu-isel < %s | FileCheck -check-prefixes=GCN-ISEL %s
+; RUN: llc -mtriple=amdgcn-amd-amdpal -mcpu=gfx900 < %s | FileCheck %s
+; Ensure that S_UADDO_PSEUDO is selected when carryout user is S_ADD_CO_PSEUDO
+
+; GCN-ISEL-LABEL: name:   s_uaddo_pseudo
+; GCN-ISEL-LABEL: body:
+; GCN-ISEL: S_UADDO_PSEUDO
+; GCN-ISEL: S_ADD_CO_PSEUDO
+
+define amdgpu_ps i32 @s_uaddo_pseudo(i32 inreg %val0, i32 inreg %val1) {
+; CHECK-LABEL: s_uaddo_pseudo:
+; CHECK:       ; %bb.0:
+; CHECK-NEXT:    s_add_i32 s0, s0, s1
```

Comment:
Not your fault but this instruction detects signed overflow, not unsigned overflow! Can you please fix this? Looks like it is original sin from https://reviews.llvm.org/D78091 @alex-t

Reply:
I'll add a separate PR to fix this after this PR is merged.   This S_ADD_I32 selection issue is hidden by bug #1.

Reply:
Actually I'll fix the S_ADD_I32 first.  I don't want to have a PR with a functionally failing compiler.

Reply:
PR to select S_ADD_U32 done in https://github.com/llvm/llvm-project/pull/160142.